### PR TITLE
Read outlines or masks from `segmentation_file_regex` parameter

### LINKED ISSUE #22: Add test and new capabilities for ALSF project
@jenna-tomkinson provided an example dataset from an ALSF project where she mentioned it would be beneficial to use CytoDataFrame. The files follow a convention as follows:
image

This issue outlines the need to add this as a test dataset and also a new capability which allows for dynamically associating filenames from original images to outline images.

As a rough sketch, we may need a new parameter which can perform regex-like associations for filenames such as:
```python
pattern_map = {
    r"CellsOutlines_BR(\d+)_C(\d+)_\d+\.tiff": r"r\1c\2f03p01-ch3sk1fk1fl1.tiff",
    r"NucleiOutlines_BR(\d+)_C(\d+)_\d+\.tiff": r"r\1c\2f03p01-ch5sk1fk1fl1.tiff"
}
```

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR adds a new parameter to CytoDataFrame: `segmentation_file_regex`. This parameter assists users in associating masks or outline files with original images where they may otherwise have different naming conventions. When `segmentation_file_regex` is None we fall back to the original capability, gracefully seeking files which match the original filenames.

Thanks for any feedback!

Closes #22 

## What kind of change(s) are included?

- [ ] Documentation (changes docs or other related content)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] These changes pass all pre-commit checks.
- [x] I have added comments to my code to help provide understanding
- [x] I have added a test which covers the code changes found within this PR
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: docs/src/examples/cytodataframe_at_a_glance.ipynb
Comment (jenna-tomkinson) [inline docs/src/examples/cytodataframe_at_a_glance.ipynb:1]: Some examples look a bit funky, especially the new ALSF data where the images are very dim. Is there any parameters that someone can use to brighten the images if they want?

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! I have a minor comment about how the new ALSF example looks and that some of the other examples look a bit funky. Maybe we can talk in person about it! Other than that, code looks pretty sound to me!

File: docs/src/examples/cytodataframe_at_a_glance.ipynb
Comment (d33bs) [reply docs/src/examples/cytodataframe_at_a_glance.ipynb:1]: Opened #28

**Comment by d33bs** [comment]
Thanks @jenna-tomkinson ! I created #28 to help address the image darkness issue and plan to meet up with you to discuss more.

**Pull request merged**